packages/sanity/src/core/releases/hooks/useDocumentVersions.tsx

```diff
@@ -33,30 +59,27 @@ const INITIAL_VALUE = {
  */
 export function useDocumentVersions(props: DocumentPerspectiveProps): DocumentPerspectiveState {
   const {documentId} = props
-
   const publishedId = getPublishedId(documentId)
-
   const documentPreviewStore = useDocumentPreviewStore()
+  const [results, setResults] = useState<DocumentPerspectiveState>(INITIAL_VALUE)
 
   const observable = useMemo(() => {
-    return documentPreviewStore
-      .unstable_observeDocumentIdSet(`sanity::versionOf("${publishedId}")`, undefined, {
-        apiVersion: RELEASES_STUDIO_CLIENT_OPTIONS.apiVersion,
-      })
-      .pipe(
-        swr(`${publishedId}`),
-        map(({value}) => ({
-          data: value.documentIds,
-          loading: false,
-          error: null,
-        })),
-        catchError((error) => {
-          return of({error, data: [] as string[], loading: false})
-        }),
-      )
+    const cachedObservable = observableCache.get(publishedId)
+    if (cachedObservable) {
+      return cachedObservable
+    }
+
+    const newObservable = createObservable(documentPreviewStore, publishedId)
+    observableCache.set(publishedId, newObservable)
```

Comment:
Ideally we should unset the cache entry when the last subscriber drops, since the way it is now, the observable will be cached forever. We can do this in a follow-up PR though.

Reply:
Cool, I've made a note about this and will come back to this 🙏